test-series/4/tools/sim_prep.py

```diff
@@ -0,0 +1,626 @@
+
+    # Inflate the simulation so antenna downselection can actually be done.
+    sim_uvd.inflate_by_redundancy()
+    
+    # Find and use the intersection of the RIMEz and H1C arrays.
+    sim_uvd = downselect_antennas(sim_uvd, ref_uvd)
+    
+    # Make sure the data is conjugated properly so redcal doesn't break.
+    sim_uvd.conjugate_bls('ant1<ant2')
+    
+    # Chunk the simulation data and write to disk.
+    chunk_sim_and_save(sim_uvd, data_files, save_dir, sky_cmp, clobber)
+    
+    return
+
+def downselect_antennas(sim_uvd, ref_uvd, tol=1.0):
+    """
+    Downselect simulated array to match unique baselines from reference.
+    
+    Parameters
+    ----------
+    sim_uvd : :class:`pyuvdata.UVData`
+        :class:`pyuvdata.UVData` object containing the simulation data.
+        
+    ref_uvd : :class:`pyuvdata.UVData`
+        :class:`pyuvdata.UVData` object containing the reference data. 
+        Only needs to have the metadata loaded.
+        
+    tol : float, optional
+        The maximum allowable discrepancy in antenna position components.
+        Default is 1 meter.
+        
+    Returns
+    -------
+    new_sim_uvd : :class:`pyuvdata.UVData`
+        Simulation data adjusted to contain a subset of the simulated array
+        such that, after appropriately translating the antenna positions, 
+        the maximum number of antennas/unique baselines remain in the 
+        intersection of the simulation subarray and the reference array. 
+        The relevant attributes of the simulation data are updated 
+        accordingly.
+
+    Notes
+    -----
+    See the memo "Antennas, Baselines, and Maximum Overlap" for details on 
+    the antenna selection process. It is worth noting that the algorithm 
+    implemented here is *not* appropriate for general arrays, but the 
+    general case should reduce to this case for an array with appropriate 
+    symmetry, as we have for the RIMEz simulation.
+    """
+    sim_uvd = _sim_to_uvd(sim_uvd)
+
+    # Find the optimal intersection, get the map between antennas, and downselect the data.
+    sim_ENU_antpos = _get_antpos(sim_uvd)
+    ref_ENU_antpos = _get_antpos(ref_uvd)
+    array_intersection = _get_array_intersection(sim_ENU_antpos, ref_ENU_antpos, tol)
+    sim_to_ref_ant_map = _get_antenna_map(array_intersection, ref_ENU_antpos, tol)
+    sim_uvd.select(antenna_nums=list(sim_to_ref_ant_map.keys()), keep_all_metadata=False)
+    ref_uvd.select(antenna_nums=list(sim_to_ref_ant_map.values()), keep_all_metadata=False)
+    
+    # Prepare the new data array.
+    new_sim_data = np.zeros_like(sim_uvd.data_array, dtype=np.complex)
+    for antpairpol in sim_uvd.get_antpairpols():
+        ai, aj, pol = antpairpol
+        ref_antpairpol = (sim_to_ref_ant_map[ai], sim_to_ref_ant_map[aj], pol)
+        blts, conj_blts, pol_inds = ref_uvd._key2inds(ref_antpairpol)
+        sim_data = sim_uvd.get_data(antpairpol)
+        if len(blts) > 0:
```

Comment:
Will do. The gist of it is that some baselines in the RIMEz simulation are conjugates of the baselines in the H1C array, so `blts` will be empty for those baselines (using `if blts:` breaks because `bool(blts)` throws an error for a nonempty `ndarray`).

Reply:
I added a comment explaining the logic. I'll hold off on marking this as resolved until I get a confirmation that the comment is OK.